### PR TITLE
4487 array out of bounds exception

### LINKED ISSUE #4487: ArrayIndexOutOfBoundsException in addMarkers(@NonNull List<MarkerOptions> markerOptionsList)
The method 

> public List<Marker> addMarkers(@NonNull List<MarkerOptions> markerOptionsList)

 in the class MapboxMap.java throws an 

> java.lang.ArrayIndexOutOfBoundsException: length=0; index=0

 although size of markerOptionsList is not 0.

My Mapbox Version is com.mapbox.mapboxsdk:mapbox-android-sdk:4.0.0-beta.2@aar

### PR DESCRIPTION
**Pull request by tobrun**
closes #4487

### CODE REVIEW

**Comment by tobrun** [comment]
cc @bleege @zugaldia

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java [hunk 1]
```diff
@@ -610,29 +610,36 @@ public Marker addMarker(@NonNull BaseMarkerOptions markerOptions) {
...
-        for (int i = 0; i < count; i++) {
-            markerOptions = markerOptionsList.get(i);
-            marker = prepareMarker(markerOptions);
-            markers.add(marker);
-        }
+        if (count > 0) {
+            MarkerOptions markerOptions;
+            Marker marker;
+            for (int i = 0; i < count; i++) {
+                markerOptions = markerOptionsList.get(i);
```
Comment (bleege) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: Would it be possible to use the type safe for loops instead?

```java
for (MarkerOptions markerOptions : markerOptionsList) {
// do stuff
}
```

**Comment by bleege** [comment]
Picking this up now.

**Comment by bleege** [comment]
Successfully tested in TestApp's Markers In Bulk activity.  Will forego loops recommendation for now to get this merged into release branch quicker so that we have more time to test the release as a whole.

**Comment by bleege** [comment]
Rebased and Merged!

**Pull request closed without merging**